### PR TITLE
Extended admin modify tutorials

### PR DESCRIPTION
**Pull request by Arcadius19**
Enhanced the administrator dashboard where the user can add a new tutorial and modify or delete any existing one.
screenshot 2
screenshot 3
screenshot 4
screenshot 5

### CODE REVIEW

File: lib/admin/admin.dart [hunk 1]
```diff
@@ -1,91 +1,65 @@
...
-    if (result != true) {
-      this.success = false;
-      print('Failed to create a new tutorial');
-    } else {
-      this.success = true;
-      this.title = null;
-      this.descEditor.getDoc().setValue('');
-      this.sourceEditor.getDoc().setValue('');
-    }
+  void updateList() {
```
Comment (NickWu007) [inline lib/admin/admin.dart]: What's this function for?

File: lib/admin/admin.html [hunk 2]
```diff
@@ -1,80 +1,28 @@
...
-    </div>
-    <div class="submit-new-tutorial-box">
-        <material-button class="create-btn" raised (trigger)="openModal()">
-            Create a tutorial
+    <div class="tl-admin-header">
+        <material-button icon class="material-drawer-button" (trigger)="drawer.toggle()">
+            <material-icon icon="menu"></material-icon>
         </material-button>
+        <img class="tl-logo" src="img/logo.svg" alt="Links Logo">
+        <div class="tl-header-name">Try Links Admin Mode</div>
```
Comment (NickWu007) [inline lib/admin/admin.html]: `Try Links`: Extra space

File: lib/app_component.dart [hunk 3]
```diff
@@ -39,6 +39,6 @@ import 'package:client/admin/admin.dart';
       path: '/tutorial/:id',
       name: 'Tutorial',
       component: TutorialPageComponent),
-  const Route(path: '/admin', name: 'Admin', component: AdminPageComponent),
+  const Route(path: '/admin/...', name: 'Admin', component: AdminPageComponent),
```
Comment (NickWu007) [inline lib/app_component.dart:42]: What's this for?

File: lib/admin/update-list-service.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import 'dart:async';
+import 'package:angular/angular.dart';
+
+@Injectable()
+class UpdateListService {
+  StreamController<bool> _updateListController = new StreamController();
+
+  UpdateListService();
+
+  Stream get updateList => _updateListController.stream;
```
Comment (NickWu007) [inline lib/admin/update-list-service.dart]: I would rename this as `updateHeadersStream`.

File: lib/admin/update-list-service.dart [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import 'dart:async';
+import 'package:angular/angular.dart';
+
+@Injectable()
+class UpdateListService {
```
Comment (NickWu007) [inline lib/admin/update-list-service.dart]: Instead of naming this as a generic stream, I would use `UpdateHeadersService`.

**Review by NickWu007 — CHANGES_REQUESTED** [review]

File: lib/admin/modify-tutorial/modify-tutorial.html [hunk 6]
```diff
@@ -0,0 +1,107 @@
+<h3 class="tutorial-panel-header">Modify a tutorial (id: {{tutorial.tutorialId}})</h3>
```
Comment (NickWu007) [inline lib/admin/modify-tutorial/modify-tutorial.html]: Don't think you need to expose the id here.

**Review by NickWu007 — CHANGES_REQUESTED** [review]

**Comment by NickWu007** [comment]
Good work Arek, just some small nits.

Btw make sure you update your local branch with the base MASTER branch. It seems like the background isn't as dim as it should be in the screenshots.

File: lib/admin/admin.dart (hunk 1 above)
Comment (Arcadius19) [reply lib/admin/admin.dart]: Right, I used it when experimenting with the shared service between the Admin Components and its children, and then forgot to delete it.

File: lib/app_component.dart (hunk 3 above)
Comment (Arcadius19) [reply lib/app_component.dart:42]: It indicated a non-terminal route and it's required for AdminPage to have child routes (/admin/add-tutorial and /modify-tutorial/:id)

**Review by NickWu007 — APPROVED** [review]

**Pull request merged**